batchflow/tests/opensets_test.py

```diff
@@ -0,0 +1,96 @@
+# pylint: disable=missing-docstring, redefined-outer-name
+
+import sys
+import pytest
+
+import PIL
+import numpy as np
+
+sys.path.insert(0, '../../')
+from batchflow.opensets import (PascalSegmentation, PascalClassification,
+                                COCOSegmentation, MNIST, CIFAR10, CIFAR100, Imagenette)
+from batchflow.opensets.ade import ADESegmentation
+
+
+@pytest.fixture()
+def pascal_segmentation():
+    return PascalSegmentation()
+
+@pytest.fixture()
+def ade_segmentation():
+    return ADESegmentation()
+
+@pytest.fixture()
+def pascal_classification():
+    return PascalClassification()
+
+@pytest.fixture()
+def coco_segmentation():
+    return COCOSegmentation()
+
+@pytest.fixture()
+def cifar10():
+    return CIFAR10()
+
+@pytest.fixture()
+def cifar100():
+    return CIFAR100()
+
+@pytest.fixture()
+def mnist():
+    return MNIST()
+
+@pytest.fixture()
+def imagenette():
+    return Imagenette()
+
+
+class TestOpensets:
+    @pytest.mark.slow
+    def test_pascal_segmentation(self, pascal_segmentation):
+        batch = pascal_segmentation.next_batch(batch_size=10)
+        assert isinstance(batch.images[0], PIL.JpegImagePlugin.JpegImageFile)
+        assert isinstance(batch.labels[0], PIL.PngImagePlugin.PngImageFile)
```

Comment:
Tests look very similar, so you can write less functions and use `pytest.mark.parametrize`

Reply:
Also, you don't need those fixtures as they are implemented now. There is no difference between them and instantiating datasets directly